File: app/src/main/java/org/keynote/godtools/android/content/ToolsCursorLoader.java
package org.keynote.godtools.android.content;

import android.content.Context;
import android.os.Bundle;

import org.ccci.gto.android.common.eventbus.content.DaoCursorEventBusLoader;
import org.cru.godtools.model.Tool;
import org.cru.godtools.model.event.content.ToolEventBusSubscriber;
import org.keynote.godtools.android.db.GodToolsDao;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class ToolsCursorLoader extends DaoCursorEventBusLoader<Tool> {
    public ToolsCursorLoader(@NonNull final Context context, @Nullable final Bundle args) {
        super(context, GodToolsDao.getInstance(context), Tool.class, args);
        addEventBusSubscriber(new ToolEventBusSubscriber(this));
    }
}

File: app/src/main/java/org/cru/godtools/content/ToolsCursorLoader.kt
package org.cru.godtools.content

import android.content.Context
import android.os.Bundle

import org.ccci.gto.android.common.eventbus.content.DaoCursorEventBusLoader
import org.cru.godtools.model.Tool
import org.cru.godtools.model.event.content.ToolEventBusSubscriber
import org.keynote.godtools.android.db.GodToolsDao

open class ToolsCursorLoader(context: Context, args: Bundle?) :
    DaoCursorEventBusLoader<Tool>(context, GodToolsDao.getInstance(context), Tool::class.java, args) {
    init {
        addEventBusSubscriber(ToolEventBusSubscriber(this))
    }
}
